core/src/main/java/org/elasticsearch/search/suggest/phrase/PhraseSuggestParser.java

```diff
@@ -324,34 +316,44 @@ private void verifyGenerator(PhraseSuggestionContext.DirectCandidateGenerator su
         }
     }
 
-    private void parseCandidateGenerator(XContentParser parser, MapperService mapperService, String fieldName,
-            PhraseSuggestionContext.DirectCandidateGenerator generator, ParseFieldMatcher parseFieldMatcher) throws IOException {
-        if (!SuggestUtils.parseDirectSpellcheckerSettings(parser, fieldName, generator, parseFieldMatcher)) {
-            if ("field".equals(fieldName)) {
-                generator.setField(parser.text());
-                if (mapperService.fullName(generator.field()) == null) {
-                    throw new IllegalArgumentException("No mapping found for field [" + generator.field() + "]");
-                }
-            } else if ("size".equals(fieldName)) {
-                generator.size(parser.intValue());
-            } else if ("pre_filter".equals(fieldName) || "preFilter".equals(fieldName)) {
-                String analyzerName = parser.text();
-                Analyzer analyzer = mapperService.analysisService().analyzer(analyzerName);
-                if (analyzer == null) {
-                    throw new IllegalArgumentException("Analyzer [" + analyzerName + "] doesn't exists");
-                }
-                generator.preFilter(analyzer);
-            } else if ("post_filter".equals(fieldName) || "postFilter".equals(fieldName)) {
-                String analyzerName = parser.text();
-                Analyzer analyzer = mapperService.analysisService().analyzer(analyzerName);
-                if (analyzer == null) {
-                    throw new IllegalArgumentException("Analyzer [" + analyzerName + "] doesn't exists");
+    static PhraseSuggestionContext.DirectCandidateGenerator parseCandidateGenerator(XContentParser parser, MapperService mapperService,
+             ParseFieldMatcher parseFieldMatcher) throws IOException {
+        XContentParser.Token token;
+        String fieldName = null;
+        PhraseSuggestionContext.DirectCandidateGenerator generator = new PhraseSuggestionContext.DirectCandidateGenerator();
+        while ((token = parser.nextToken()) != Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                fieldName = parser.currentName();
+            }
+            if (token.isValue()) {
+                if (!SuggestUtils.parseDirectSpellcheckerSettings(parser, fieldName, generator, parseFieldMatcher)) {
+                    if ("field".equals(fieldName)) {
```

Comment:
I wasn't going to change this in the PhraseSuggestParser since this is going to eventually be replaced by the fromXContent() method in PhraseSuggestBuilder. But I will use ParseFields there.